feat!: update middleware contracts version (#585)
Layr-Labs/eigensdk-go: CHANGELOG.md

Reviewer: Let's mention `egnaddrs` was updated to work with slashing AVSs and now returns `0x00..00` when it fails to fetch the ServiceManager from the RegistryCoordinator. Also, can we add some of this information to the command's "help" documentation?

Author: Done in [6649ce8](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/6649ce86ca9cd78907521a2e8165b0c5444a13b4) and [6f2b69c](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/6f2b69ca8b0f9c15aa1004bd43c753e0e6c30db3)

* Added field `DontUseAllocationManager` to `BuildAllConfig` by @MegaRedHand in [#580](https://github.com/Layr-Labs/eigensdk-go/pull/580)
* Added `AnvilC` field to `clients.Clients` struct by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)

> * Added `IsOperatorRegisteredWithAvs`, `GetAVSRegistrar` methods to elcontracts chain reader and and `SetAVSRegistrar` to chain writer by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)

Reviewer: Let's add some examples here

Author: Done in [82d0647](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/82d0647aff916052767b8e0d4d50a5b43aad84ea)

    ).WithWindowDuration(windowDuration)
    blsAggServ.InitializeNewTask(metadata)
    ```

> * fix: `IsOperatorRegisteredWithOperatorSet` only queries to AVSDirectory by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)

Reviewer: Let's mention `IsOperatorRegisteredWithAvs` as an alternative for M2 quorum users.

Author: Done in [63ef39f](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/63ef39fd5a2a750eea5ee3794b14858d69cc25c8)

    ).WithWindowDuration(windowDuration)
    blsAggServ.InitializeNewTask(metadata)
    ```

> * The elcontracts `chainReader.IsOperatorRegisteredWithOperatorSet` does not query to AVSDirectory anymore, only to allocationManager by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)

Reviewer: ```suggestion
* In elcontracts, `ChainReader.IsOperatorRegisteredWithOperatorSet` no longer queries the `AVSDirectory`. It now only queries the `AllocationManager` by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
```

Author: Done in [1412b2c](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/1412b2cd0c6afe3df2e3340503cabe669b52260f)

>   * The query to AVSDirectory i still available using `chainReader.IsOperatorRegisteredWithAvs` method

Reviewer: ```suggestion
  * The query to AVSDirectory is still available using `chainReader.IsOperatorRegisteredWithAvs` method
```

Author: Done in [1412b2c](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/1412b2cd0c6afe3df2e3340503cabe669b52260f)

    ).WithWindowDuration(windowDuration)
    blsAggServ.InitializeNewTask(metadata)
    ```
* In elcontracts, `ChainReader.IsOperatorRegisteredWithOperatorSet` no longer queries the `AVSDirectory`. It now only queries the `AllocationManager` by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)

>   * The query to AVSDirectory is still available using `chainReader.IsOperatorRegisteredWithAvs` method

Reviewer: Let's rewrite this to mention `IsOperatorRegisteredWithOperatorSet` only works for operator sets, while `IsOperatorRegisteredWithAvs` works for m2 quorums

Author: Done in [3e2b71e](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/3e2b71e9eac5e83856dbbd9aba8c27a1975024b0)

    ).WithWindowDuration(windowDuration)
    blsAggServ.InitializeNewTask(metadata)
    ```

> * In elcontracts, `ChainReader.IsOperatorRegisteredWithOperatorSet` no longer queries the `AVSDirectory`, only `AllocationManager`, since only works for operator sets by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
>   * To query if an operator is register to an m2 quorum you should now use `chainReader.IsOperatorRegisteredWithAvs` method, that queries the AVSDirectory.
> * `egnaddrs` utility now works with slashing version, and in case the registryCoordinator is the slashing version and service manager address is not passed by flag, then the returned service manager address will be address zero (0x00..00)

Reviewer: ```suggestion
* In elcontracts, `ChainReader.IsOperatorRegisteredWithOperatorSet` no longer queries the `AVSDirectory`, and so now only works for operator sets by @maximopalopoli in [#585](https://github.com/Layr-Labs/eigensdk-go/pull/585)
  * To query if an operator is registered to an M2 quorum you should now use `chainReader.IsOperatorRegisteredWithAvs`, which queries the `AVSDirectory`.
* `egnaddrs` utility now works with slashing release middleware contracts and, in that case, the returned service manager will be the zero address, unless the `--service-manager` is specified.
```
